UnityProject/Assets/Scripts/UI/Action/RequestGameActionSO.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using UnityEngine;
+using Utility = UnityEngine.Networking.Utility;
+using Mirror;
+
+public class RequestGameActionSO : ClientMessage
+{
+	public static short MessageType = (short)MessageTypes.RequestGameActionSO;
+	public string soName;
+
+	public override IEnumerator Process()
```

Comment:
Oh cool, this allows you to request an action via an SO instead of via a component, so it looks like you added multiple ways to trigger UI actions.

One issue, sending a string name over the network is pretty expensive in terms of bandwidth. Ideally it should be a numeric ID. There are a few suggestions online for generating unique IDs for scriptable objects, such as doing it when the asset is saved.

EDIT: In fact, I saw that you already have setup logic for these SO's so actually generating the ID's is really straightforward! You can use a similar approach as with the component IDs - just sort the SO assets alphabetically by name, and us an auto-incrementing ushort as the id. You can build the dict that way instead of using string as the key.